src/libs/Timers.js

```diff
@@ -0,0 +1,31 @@
+import _ from 'underscore';
+
+const timers = [];
+
+/**
+ * Register a timer so it can be cleaned up later.
+ *
+ * @param {Number} timerID
+ * @returns {Number}
+ */
+function register(timerID) {
+    timers.push(timerID);
+    return timerID;
+}
+
+/**
+ * Clears all timers that we have registered. Use for long running tasks that may begin once logged out.
+ */
+function clearAll() {
+    _.each(timers, (timer) => {
+        // We don't know whether it's a setTimeout or a setInterval, but it doesn't really matter. If the id doesn't
+        // exist nothing bad happens.
+        clearTimeout(timer);
+        clearInterval(timer);
+    });
+}
```

Comment:
`const timers = [];` is never emptied

`clearAll` will be foreach-ing already cleared timers

Maybe it can be refactored to something like 

```js
const timers = {};

function clearSingle(id) {
  clearTimeout(id);
  clearInterval(id);
  delete timers[id];
}

function clearAll() {
  _.each(timers, clearSingle);
}
```

Reply:
That's a good point. But there is no danger in clearing already cleared timers so this might never be a problem.